packages/republik-crowdfundings/lib/scheduler/importPayments.ts

```diff
@@ -37,32 +34,45 @@ export async function importPayments(
   _args: undefined,
   context: Context,
 ): Promise<void> {
-  const transaction = await context.pgdb.transactionBegin()
-  let insertPaymentReport, unmatchedPaymentsAmount, matchingReport
-
   try {
-    await getAndInsertPostfinanceImports(transaction)
-    insertPaymentReport = await insertNewPayments(transaction)
-    matchingReport = await tryMatchingPayments(transaction, context)
-    await transaction.transactionCommit()
-    unmatchedPaymentsAmount = await getUnmatchedPaymentsAmount(context.pgdb)
+    await withTransaction(syncPaymentFiles, context.pgdb)
+    await withTransaction(_importPayments(context), context.pgdb)
+    await sendReminders()
   } catch (e) {
-    transaction.transactionRollback()
-    informFailed(
+    await informFailed(
       `importPayments(): postfinance sync failed with the following error: ${
         e.message || e
       } ${e.stack || ''}`,
     )
     return
   }
+}
+
+const _importPayments = (context: Context) => async (transaction: PgDb) => {
```

Comment:
maybe `createPaymentImporter` would be a good name (since the function is a factory)